packages/x-tailwindcss/src/x-tailwind-plugin/components/input/line.ts

```diff
@@ -0,0 +1,33 @@
+import { TailwindHelpers } from '../../../types';
+
+/**
+ * Returns the `line` variant for component `input`.
+ *
+ * @param helpers - The {@link TailwindHelpers} to generate CSS.
+ * @returns The {@link CssStyleOptions} for the variant.
+ */
+// eslint-disable-next-line  @typescript-eslint/explicit-function-return-type
+export function inputLine(helpers: TailwindHelpers) {
+  const { theme } = helpers;
+  return {
+    line: {
+      borderWidth: 0,
+      borderBottomWidth: theme('borderWidth.1'),
+
+      '&:hover': {
+        borderBottomWidth: theme('borderWidth.2')
+      },
+
+      '&:focus': {
+        borderColor: theme('colors.neutral.90'),
+        color: theme('colors.neutral.90'),
+        borderBottomWidth: theme('borderWidth.2')
+      },
+
+      '&:disabled': {
+        backgroundColor: theme('colors.neutral.10'),
+        borderColor: theme('colors.neutral.25')
+      }
```

Comment:
To avoid size sifting, we can use the background as bottom line:

```suggestion
      '--border-color': 'var(--input-color-50)',
      borderWidth: 0,
      backgroundImage: 'linear-gradient(var(--border-color) ,var(--border-color))',
      backgroundPosition: 'bottom',
      backgroundSize: '100% 1px',
      backgroundRepeat: 'no-repeat',

      '&:hover': {
        backgroundSize: '100% 2px'
      },

      '&:focus': {
        backgroundSize: '100% 2px',
        color: theme('colors.neutral.90')
      },

      '&:disabled': {
        backgroundColor: theme('colors.neutral.10'),
        '--border-color': theme('colors.neutral.25')
      }
```